### ISSUE TITLE
Importing aws_ssoadmin_account_assignment always causes destroy and recreate

### ISSUE THREAD

**Issue by mathewpeterson** [hashicorp/terraform-provider-aws#17179]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.14.4
+ provider registry.terraform.io/hashicorp/aws v3.24.1

### Affected Resource(s)

* aws_ssoadmin_account_assignment

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl

data "aws_organizations_organization" "this" {}

data "aws_ssoadmin_instances" "this" {}

data "aws_identitystore_group" "devops" {
  identity_store_id = tolist(data.aws_ssoadmin_instances.this.identity_store_ids)[0]
  filter {
    attribute_path  = "DisplayName"
    attribute_value = "Dev Ops"
  }
}

resource "aws_ssoadmin_permission_set" "administrator" {
    name             = "AdministratorAccess"
    description      = "AWS managed IAM Policy "
    instance_arn     = tolist(data.aws_ssoadmin_instances.this.arns)[0]
    session_duration = "PT1H"
    tags             = {}
}

resource "aws_ssoadmin_account_assignment" "devops_administrator" {
    for_each = toset(["1234567890"])

    instance_arn       = aws_ssoadmin_permission_set.administrator.instance_arn
    permission_set_arn = aws_ssoadmin_permission_set.administrator.arn
    principal_id       = data.aws_identitystore_group.devops.group_id
    principal_type     = "GROUP"
    target_type        = "AWS_ACCOUNT"
    target_id          = each.key
}
```

### Expected Behavior

If I import this resource using the composite key, then when I run `terraform plan` I would expect Terraform to report `No changes. Infrastructure is up-to-date.` Also when you allow Terraform to proceed, it ends up failing with `error creating SSO Account Assignment for GROUP (0000000000-00000000-0000-0000-0000-000000000000): already exists` but that might be a unrelated issue because if you just re-run then it seems to create the resource.

### Actual Behavior

```
  # aws_ssoadmin_account_assignment.devops_administrator[1234567890] will be destroyed
  - resource "aws_ssoadmin_account_assignment" "devops_administrator" {
      - id                 = "0000000000-00000000-0000-0000-0000-000000000000,GROUP,1234567890,AWS_ACCOUNT,arn:aws:sso:::permissionSet/ssoins-0000000000000000/ps-0000000000000000,arn:aws:sso:::instance/ssoins-0000000000000000" -> null
      - instance_arn       = "arn:aws:sso:::instance/ssoins-0000000000000000" -> null
      - permission_set_arn = "arn:aws:sso:::permissionSet/ssoins-0000000000000000/ps-0000000000000000" -> null
      - principal_id       = "0000000000-00000000-0000-0000-0000-000000000000," -> null
      - principal_type     = "GROUP" -> null
      - target_id          = "1234567890" -> null
      - target_type        = "AWS_ACCOUNT" -> null
    }

  # aws_ssoadmin_account_assignment.devops_administrator["1234567890"] will be created
  + resource "aws_ssoadmin_account_assignment" "devops_administrator" {
      + id                 = (known after apply)
      + instance_arn       = "arn:aws:sso:::instance/ssoins-0000000000000000"
      + permission_set_arn = "arn:aws:sso:::permissionSet/ssoins-0000000000000000/ps-0000000000000000"
      + principal_id       = "0000000000-00000000-0000-0000-0000-000000000000"
      + principal_type     = "GROUP"
      + target_id          = "1234567890"
      + target_type        = "AWS_ACCOUNT"
    }
```

### Steps to Reproduce

1. Have existing AWS SSO permission sets and account assignments
2. Import the existing assignments 
3. Run `terraform plan` to see that Terraform wants to `destroy` and then `create` the resource.

### Important Factoids

I suspect there are issues with the import code where it is not persisting the `index_key` as a string but rather a number in the state file.

**Comment by anGie44** [comment]
Hi @mathewpeterson, thank you for raising this issue. 

Looking at the `terraform plan` output, i do see the indices of the resource after import are different  `aws_ssoadmin_account_assignment.devops_administrator[1234567890]` (the index i.e. account # is represented as an `int`) 
vs. 
`aws_ssoadmin_account_assignment.devops_administrator["1234567890"]` (the index i.e. account # is represented as a `string`), 
which I believe explains the destroy/create before. did you by chance use the `int` value during import time such as:
```
terraform import 'aws_ssoadmin_account_assignment.devops_administrator[1234567890]' 
```
?

I believe in this case, wrapping that account number with quotes during the `import` should later show `No changes...` when running `terraform plan` as the `for_each` uses the `string` representation of the `target_id`. This documentation here shows a couple examples when using `for_each`: https://www.terraform.io/docs/commands/import.html#example-import-into-resource-configured-with-for_each

**Comment by mathewpeterson** [comment]
Hm, yes I believe you are correct. I did not have to wrap the entire thing around single quotes, in fact Terraform produced an error, but wrapping the account number in double quotes seemed to do the trick. 

I had written a bash script to enumerate through a couple dozen aws account numbers to import the resources and bash had dropped/evaluated the double quotes so I had to escape them.

This can be closed as I believe this is really an issue around Terraform core. 

Thank you so much for all your help!
